src/Umbraco.Infrastructure/Configuration/JsonConfigManipulator.cs

```diff
@@ -216,8 +245,13 @@ private static void RemoveJsonKey(JObject? json, string key)
             token?.Parent?.Remove();
         }
 
-        private void SaveJson(JsonConfigurationProvider provider, JObject? json)
+        private void SaveJson(JsonConfigurationProvider? provider, JObject? json)
```

Comment:
If you put `(_configuration as IConfigurationRoot)?.Reload();` here it doesn't need to be in all the other methods.

Reply:
The class JsonConfigManipulator should reload configuration without manipulating json config, gotya!

Reply:
We should probably split updating `IConfiguration` and the JSON file indeed, but in that case, it's a better idea to add our own `IConfigurationProvider` that persists the value(s) and directly reloads the configuration.